osu.Game/Rulesets/Edit/HitObjectComposer.cs

```diff
@@ -531,6 +531,9 @@ private void toolSelected(CompositionTool tool)
 
             if (!(tool is SelectTool))
                 EditorBeatmap.SelectedHitObjects.Clear();
+
+            // Prevent object context menu from staying open.
+            GetContainingFocusManager()?.ChangeFocus(null);
```

Comment:
This is dodgy as all hell, and may be broken by future changes that allow multiple simultaneously focused drawables (hi @smoogipoo).

I think we need a proper "close the context menu that I opened" primitive in framework. Maybe something like (***VERY UNTESTED***, written just to demonstrate general idea):

```diff
diff --git a/osu.Framework/Graphics/Cursor/ContextMenuContainer.cs b/osu.Framework/Graphics/Cursor/ContextMenuContainer.cs
index 9ff908d69..0174322cf 100644
--- a/osu.Framework/Graphics/Cursor/ContextMenuContainer.cs
+++ b/osu.Framework/Graphics/Cursor/ContextMenuContainer.cs
@@ -76,6 +76,14 @@ protected override bool OnMouseDown(MouseDownEvent e)
 
                     menuTarget = target;
 
+                    menuTarget.CloseRequested += t =>
+                    {
+                        if (menuTarget != t)
+                            return;
+
+                        menu.Close();
+                    };
+
                     if (menuTarget == null || items.Length == 0)
                     {
                         if (menu.State == MenuState.Open)
diff --git a/osu.Framework/Graphics/Cursor/IHasContextMenu.cs b/osu.Framework/Graphics/Cursor/IHasContextMenu.cs
index 0a214a2f0..c3188e873 100644
--- a/osu.Framework/Graphics/Cursor/IHasContextMenu.cs
+++ b/osu.Framework/Graphics/Cursor/IHasContextMenu.cs
@@ -1,6 +1,7 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
 using osu.Framework.Graphics.UserInterface;
 
 namespace osu.Framework.Graphics.Cursor
@@ -15,5 +16,9 @@ public interface IHasContextMenu : IDrawable
         /// <para>If null, this <see cref="Drawable"/> will not be picked as the menu target and other <see cref="Drawable"/>s underneath may become the menu target.</para>
         /// </remarks>
         MenuItem[]? ContextMenuItems { get; }
+
+        internal Action<IHasContextMenu>? CloseRequested { get; set; }
+
+        public void RequestClose() => CloseRequested?.Invoke(this);
     }
 }

```